portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,74 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Noelle!</h1>");
+        private final List<String> commStream = new ArrayList<>();
+
+		@Override
+		public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+			Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+
+            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+            PreparedQuery results = datastore.prepare(query);
+
+            List<String> tasks = new ArrayList<>();
+            for (Entity entity : results.asIterable()) {
+             long id = entity.getKey().getId();
+                String comm = (String) entity.getProperty("content");
+                
+                tasks.add(comm);
+                
+
+            }
+
+			response.setContentType("application/json;");
+			String json = new Gson().toJson(tasks);
+			System.out.println("get: " + tasks);
+			response.getWriter().println(json);
+		}
+        
+        @Override
+		public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+			String text =  getParameter(request, "textarea_field", "");//returns what is in text firld
+			
+            commStream.add(text);
+			System.out.println("post: " + commStream);
```

Comment:
I believe this is just for debugging, is it? If it's just for debugging and you confirm that the feature is working on the page now, you can delete these two lines (and also delete private final List<String> commStream = new ArrayList<>(); if this variable is no longer used)